src/mlpack/core/data/csv_parser_impl.hpp

```diff
@@ -0,0 +1,257 @@
+
+    if ((N == 3) || (N == 4))
+    {
+      const bool neg = (str[0] == '-');
+      const bool pos = (str[0] == '+');
+
+      const size_t offset = ((neg || pos) && (N == 4)) ? 1 : 0;
+
+      const char sig_a = str[offset];
+      const char sig_b = str[offset+1];
+      const char sig_c = str[offset+2];
+
+      if (((sig_a == 'i') || (sig_a == 'I')) &&
+          ((sig_b == 'n') || (sig_b == 'N')) &&
+	  ((sig_c == 'f') || (sig_c == 'F')))
+      {
+        val = neg ? -(std::numeric_limits<typename MatType::elem_type>::infinity()) :
+	            std::numeric_limits<typename MatType::elem_type>::infinity();
+        return true;
+      }
+      else if (((sig_a == 'n') || (sig_a == 'N')) &&
+	       ((sig_b == 'a') || (sig_b == 'A')) &&
+	       ((sig_c == 'n') || (sig_c == 'N')))
+      {
+        val = std::numeric_limits<typename MatType::elem_type>::quiet_NaN();
+        return true;
+      }
+    }
+
+    char* endptr = nullptr;
+
+    if ((std::is_floating_point<typename MatType::elem_type>::value) ||
+        (std::is_integral<typename MatType::elem_type>::value))
+    {
+      val = typename MatType::elem_type(std::strtod(str, &endptr));
+    }
+    else
+    {
+      if (std::is_signed<typename MatType::elem_type>::value)
+      {
+        val = typename MatType::elem_type(std::strtoll(str, &endptr, 10));
+      }
+      else
+      {
+        if (str[0] == '-')
+        {
+          val = typename MatType::elem_type(0);
+          return true;
+        }
+
+        val = typename MatType::elem_type( std::strtoull(str, &endptr, 10));
+      }
+    }
+
+    if (str == endptr)
+    {
+      return false;
+    }
+    return true;
+  }
+
+  template<typename MatType>
+  bool LoadCSVV(MatType& x, std::fstream& f)
+  {
+    bool load_okay = f.good();
+
+    f.clear();
+    
+    const std::fstream::pos_type pos1 = f.tellg();
+    
+    size_t f_n_rows = 0;
+    size_t f_n_cols = 0;
+    
+    std::string line_string;
+    std::stringstream line_stream;
+    std::string token;
+
+    while (f.good() && load_okay)
+    {
+      std::getline(f, line_string);
+      if (line_string.size() == 0)
+      {
+        break;
+      }
+      line_stream.clear();
+      line_stream.str(line_string);
+
+      size_t line_n_cols = 0;
+
+      while (line_stream.good())
+      {
+        std::getline(line_stream, token, ',');
+        ++line_n_cols;
+      }
+
+      if (f_n_cols < line_n_cols)
+      {
+        f_n_cols = line_n_cols;
+      }
+
+      ++f_n_rows;
+    }
+
+    f.clear();
+    f.seekg(pos1);
+
+    x.zeros(f_n_rows, f_n_cols);
+
+    size_t row = 0;
+
+    while (f.good())
+    {
+      std::getline(f, line_string);
+
+      if (line_string.size() == 0)
+      {
+        break;
+      }
+
+      line_stream.clear();
+      line_stream.str(line_string);
+
+      size_t col = 0;
+
+      while (line_stream.good())
+      {
+        std::getline(line_stream, token, ',');
+        ConvertToken<MatType>(x.at(row, col), token);
```

Comment:
When we try to load and have a `DatasetMapper<>`, here is where we should call `MapString()` when `ConvertToken()` returns `false`. :+1: